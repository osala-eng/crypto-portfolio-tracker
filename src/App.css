.App {
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

#dashboard_heading{
  margin-top: 150px;
  text-align: left;
  font-size: 30px;
  font-weight: bold;
  height: fit-content;
}

#dashboard-container{
  display: flex;
  flex-direction: column;
  width: 90%;
  align-self: center;
}