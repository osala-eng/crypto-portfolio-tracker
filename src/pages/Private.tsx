import React from 'react';
import {Navigate} from 'react-router-dom';

const Private = ({children, access}: {
    children: JSX.Element,
    access?: boolean
}) =>  access ? children : <Navigate to={'/'}/>;

export default Private;
