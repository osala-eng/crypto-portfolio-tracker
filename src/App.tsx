import React, { useState } from 'react';
import { Route, Routes, HashRouter } from 'react-router-dom';
import Dashbord from './pages/Dashboard';
import Home from './pages/Home';
import Private from './pages/Private';


function App() {
  const [auth, setAuth] = useState(false);
  const [user, setuserName] = useState('JohnW');
  /* istanbul ignore next */
  const authenticate = (User: string) => {
    setAuth(true);
    setuserName(User);
  };
  return (
    <HashRouter>
      <Routes>
        <Route path='/' element={<Home authenticate={authenticate}/>}/>
        <Route path='/dashboard' element={
          <Private access={auth}>
            <Dashbord user={user}/>
          </Private>
        }/>
      </Routes>
    </HashRouter>
  );
};

export default App;
