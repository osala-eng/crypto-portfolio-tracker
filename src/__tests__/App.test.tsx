import React, { useEffect } from 'react';
import App from '../App';
import renderer from 'react-test-renderer';
import Private from '../pages/Private';
import {HashRouter, Routes, Route, useNavigate} from 'react-router-dom';
import Dashbord from '../pages/Dashboard';


const func = jest.fn();

test('All snapshots match tree', () => {
    const element = renderer.create(<App />);
    const tree = element.toJSON();
    expect(tree).toMatchSnapshot();
});

const TestRoot = ()=>{
    const navigate = useNavigate();
    return(
        <div>
            <button onClick={()=>{navigate('/test1')}}>Test1</button>
            <button onClick={()=>{navigate('/test0')}}>Test0</button>
        </div>
    )
}

const RoutedElm0 = <HashRouter>
    <Routes>
        <Route path='/' element={
            <Private access={true}>
                <Dashbord  user='JohnW'/>
            </Private>
        }/>
    </Routes>
</HashRouter>;

describe('Test private render 1', ()=>{
    test('Compare snapshot 1', () => {
        const element = renderer.create(RoutedElm0);
        const tree = element.toJSON();
        expect(element).toMatchSnapshot();
    });
});



const RoutedElm1 = <HashRouter>
    <Routes>
        <Route path='/' element = {
            <Private>
                <></>
            </Private>
        }/>
    </Routes>
</HashRouter>;

describe('Test private render  2', ()=>{
    test('Compare snapshot 2', () => {
        const element = renderer.create(RoutedElm1);
        const tree = element.toJSON();
        expect(element).toMatchSnapshot();
    });
});
