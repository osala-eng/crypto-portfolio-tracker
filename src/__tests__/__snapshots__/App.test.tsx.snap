// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All snapshots match tree 1`] = `
<div
  className="App"
>
  <div
    className="logo-container"
    id="logo_disp"
  >
    <a
      href="#/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        height={150}
        id="logo"
        src="logo.png"
        width={150}
      />
    </a>
    <div
      id="name_box"
    >
      <h1
        id="company_name"
      >
        Crypto Tracker
      </h1>
    </div>
  </div>
  <div
    id="product_headline"
  >
    <div
      id="prod_head_container"
    >
      <h1
        className="crypto-headline"
      >
        Crypto Tracker: All your crypto in one place
      </h1>
      <p
        id="product_description"
      >
        Crypto Tracker is a web app that allows you to easily manage your Crypto currency holdings in one place. Keep track of the prices and your profit/loss trends.
      </p>
    </div>
  </div>
  <div
    id="features-container"
  >
    <h1
      id="features_heading"
    >
      Features
    </h1>
    <div
      id="features_list"
    >
      <div
        className="feature-container"
      >
        <p
          className="feature"
        >
          Add/Remove Crypto Assets
        </p>
      </div>
      <div
        className="feature-container"
      >
        <p
          className="feature"
        >
          Track Prices
        </p>
      </div>
      <div
        className="feature-container"
      >
        <p
          className="feature"
        >
          View PnL
        </p>
      </div>
    </div>
  </div>
  <div
    id="parent_container_login"
  >
    <div
      id="register-container"
    >
      <h1
        id="registration_heading"
      >
        Registration
      </h1>
      <form
        id="registration_form"
      >
        <div
          className="user-input"
        >
          <label
            className="box-label"
            htmlFor="username_field"
          >
            username
          </label>
          <input
            className="box-input"
            id="username_field"
            name="username_field"
            onChange={[Function]}
            type="text"
          />
        </div>
        <div
          className="user-input"
        >
          <label
            className="box-label"
            htmlFor="email_field"
          >
            email
          </label>
          <input
            className="box-input"
            id="email_field"
            name="email_field"
            onChange={[Function]}
            type="email"
          />
        </div>
        <div
          className="user-input"
        >
          <label
            className="box-label"
            htmlFor="password_field"
          >
            password
          </label>
          <input
            className="box-input"
            id="password_field"
            name="password_field"
            onChange={[Function]}
            type="password"
          />
        </div>
        <div
          id="button_container"
        >
          <button
            id="signup_button"
            onClick={[Function]}
            type="submit"
          >
            Signup
          </button>
        </div>
      </form>
    </div>
    <div
      id="register-container"
    >
      <h1
        id="login_heading"
      >
        Login
      </h1>
      <form
        id="login_form"
      >
        <div
          className="user-input"
          id="user-input-div"
        >
          <label
            className="box-label"
            htmlFor="login_username_field"
          >
            username
          </label>
          <input
            className="box-input"
            data-testid="login-user"
            id="login_username_field"
            name="login_username_field"
            onChange={[Function]}
            type="text"
          />
        </div>
        <div
          className="user-input"
        >
          <label
            className="box-label"
            htmlFor="login_password_field"
          >
            password
          </label>
          <input
            className="box-input"
            data-testid="login-pass"
            id="login_password_field"
            name="login_password_field"
            onChange={[Function]}
            type="password"
          />
        </div>
        <div
          id="button_container"
        >
          <button
            data-testid="login-btn"
            id="login_button"
            onClick={[Function]}
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
  <footer
    id="footer_container"
  >
    <div
      className="footer-div"
    />
    <div
      className="footer-div"
      id="copyright_notice"
    >
      Copyright Crypto Tracker
    </div>
    <div
      className="footer-div"
      id="sm_links"
    >
      <a
        className="image-icos"
        href="//skillreactor.io"
        id="sm_facebook"
      >
        <img
          alt="facebook icon"
          height={30}
          src="facebook.png"
          width={30}
        />
      </a>
      <a
        className="image-icos"
        href="//skillreactor.io"
        id="sm_linkedin"
      >
        <img
          alt="linkedin icon"
          height={30}
          src="linkedin.png"
          width={30}
        />
      </a>
      <a
        className="image-icos"
        href="//skillreactor.io"
        id="sm_twitter"
      >
        <img
          alt="twitter icon"
          height={30}
          src="twitter.png"
          width={30}
        />
      </a>
    </div>
  </footer>
</div>
`;

exports[`Test private render  2 Compare snapshot 2 1`] = `null`;

exports[`Test private render 1 Compare snapshot 1 1`] = `
<div
  className="App"
>
  <div
    className="logo-container"
    id="logo_disp"
  >
    <a
      href="#/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        height={150}
        id="logo"
        src="logo.png"
        width={150}
      />
    </a>
    <div
      id="name_box"
    >
      <h1
        id="company_name"
      >
        Crypto Tracker
      </h1>
    </div>
  </div>
  <div
    id="dashboard-container"
  >
    <div
      id="dashboard_heading"
    >
      Dashboard
    </div>
    <table
      id="table-container"
    >
      <thead>
        <tr>
          <th
            className="table_heading"
          >
            Token
          </th>
          <th
            className="table_heading"
          >
            Qty. Owned
          </th>
          <th
            className="table_heading"
          >
            Price
          </th>
          <th
            className="table_heading"
          >
            Total Value
          </th>
          <th
            className="table_heading"
          >
            Allocation
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="table-row-class"
        >
          <td
            className="table_data"
          >
            bitcoin
          </td>
          <td
            className="table_data"
          >
            0.5
          </td>
          <td
            className="table_data"
          >
            10000
          </td>
          <td
            className="table_data"
          >
            5000
          </td>
          <td
            className="table_data"
          >
            50%
          </td>
        </tr>
        <tr
          className="table-row-class"
        >
          <td
            className="table_data"
          >
            ethereum
          </td>
          <td
            className="table_data"
          >
            25
          </td>
          <td
            className="table_data"
          >
            200
          </td>
          <td
            className="table_data"
          >
            5000
          </td>
          <td
            className="table_data"
          >
            50%
          </td>
        </tr>
      </tbody>
    </table>
    <form
      id="crypto-input-form"
    >
      <div
        className="crypto-input-field"
      >
        <label
          className="crypto-input-label-class"
          htmlFor="dashboard_token"
        >
          Token
        </label>
        <input
          className="crypto-text-input-class"
          id="dashboard_token"
          type="text"
        />
      </div>
      <div
        className="crypto-input-field"
      >
        <label
          className="crypto-input-label-class"
          htmlFor="dashboard_quantity"
        >
          Qty. Owned
        </label>
        <input
          className="crypto-text-input-class"
          id="dashboard_quantity"
          type="text"
        />
      </div>
      <div
        id="crypto-button-container"
      >
        <button
          id="dashboard_add_button"
          type="submit"
        >
          Add Asset
        </button>
      </div>
    </form>
  </div>
</div>
`;
